lib/rubocop/rspec/language/node_pattern.rb

```diff
@@ -19,11 +20,35 @@ module NodePattern
 
         def_node_matcher :example?, Examples::ALL.block_pattern
 
-        def_node_matcher :hook?, Hooks::ALL.block_pattern
-
         def_node_matcher :let?, Helpers::ALL.block_or_block_pass_pattern
 
         def_node_matcher :subject?, Subject::ALL.block_pattern
+
+        def_runtime_node_matcher :hook? do
+          hook_selectors.block_pattern
+        end
+
+        def hook_selectors
+          custom_method_names = config
+            .for_all_cops
+            .fetch('RSpec', {})
```

Comment:
This could be `.dig('RSpec', 'Language', 'Hooks').to_a`

We don't need to calculate this every time however, only once, when reading the config. So it still could be a constant

Reply:
Yeah, I'll add some config access method once there will be more language elements implemented.